env/fs_posix.cc

```diff
@@ -620,9 +620,10 @@ class PosixFileSystem : public FileSystem {
       }
     }
 
-    const auto pre_read_errno = errno;  // errno may be modified by readdir
+    // reset errno before calling readdir()
+    errno = 0;
     struct dirent* entry;
-    while ((entry = readdir(d)) != nullptr && errno == pre_read_errno) {
```

Comment:
Since `readdir` may get called multiple times, you will need to reset `errno` every time before you call `readdir`.

Reply:
Just curious: is it possible that return value of readdir is not null and error is set? Considering errno is thread local.

Reply:
It may be possible. However, if the return value of `readdir` is not `NULL` then we should treat `errno` value as `undefined`.